packages/core-p2p/lib/monitor.js

```diff
@@ -413,25 +399,29 @@ class Monitor {
 
   /**
    * Populate list of available peers from random peers.
-   * @return {Peer[]}
    */
   async discoverPeers() {
-    try {
-      const peers = await this.getRandomPeer().getPeers()
-
-      peers.forEach(peer => {
-        if (
-          Peer.isOk(peer) &&
-          !this.getPeer(peer.ip) &&
-          !this.guard.isMyself(peer)
-        ) {
-          this.__addPeer(peer)
+    const shuffledPeers = shuffle(this.getPeers())
+    const nNewPeersNeeded = 4
```

Comment:
why 4?

Reply:
It was an arbitrary number to fix this deficiency in `discoverPeers()` before the patch:
`discoverPeers()` would be satisfied as long as it fetches all peers from some of its peers. E.g.
ME <-> P (P is my peer). If I fetch all P's peers then it is ok and `discoverPeers()` considers the job done and quits. However it could be that the list of P's peers is quite short - maybe just 1 or 2 entries.

To fix this I added this magic number "4" - at least 4 new peers must be added, even if we have to cycle through more than one of our peers (e.g. fetch 2 from P and 2 others from Q).

However, this can be done better - I changed the code to consider the job done and quit once `this.hasMinimumPeers()` becomes true.